### ISSUE TITLE
 contraint violation cause of JdbcOperationsSessionRepository.JdbcSession#setAttribute

### ISSUE THREAD

**Issue by mfra** [spring-projects/spring-session#1110]
Removing and readding a attribute within a single requests causes a DB's contraint violation.

This is caused by the session's attribute delta calculation in
`org.springframework.session.jdbc.JdbcOperationsSessionRepository.JdbcSession#setAttribute`

Removing the attribute stores "attributeName:REMOVED" while adding later changes the information to "attributeName:ADDED". The DELETE  statement then is not executed and an INSERT of that attribute is triggered on the end of the requestcycle. This leads to a constraint violation defined by 
` CONSTRAINT SPRING_SESSION_ATTRIBUTES_PK PRIMARY KEY (SESSION_PRIMARY_ID, ATTRIBUTE_NAME)`

**Comment by vpavic** [comment]
Thanks for the report @mfra. Can you clarify which version of Spring Session are you using?

**Comment by mfra** [comment]
I am using currently:
```
  <groupId>org.springframework.session</groupId>
  <artifactId>spring-session-jdbc</artifactId>
  <version>2.0.3.RELEASE</version>
```

**Comment by vpavic** [comment]
Thanks for the quick follow up. Please upgrade to `2.0.4.RELEASE` - this issue was reported and addressed in #1070.

**Comment by vpavic** [comment]
Duplicate of #1070
